test/Microsoft.AspNetCore.Server.Kestrel.Transport.Libuv.Tests/ConnectionTests.cs

```diff
@@ -16,38 +18,86 @@ public class ConnectionTests
-            {
-                var mockLibuv = new MockLibuv();
-                var transportContext = new TestLibuvTransportContext() { ConnectionHandler = mockConnectionHandler };
-                var transport = new LibuvTransport(mockLibuv, transportContext, null);
-                var thread = new LibuvThread(transport);
+            var mockConnectionHandler = new MockConnectionHandler();
+            var mockLibuv = new MockLibuv();
+            var transportContext = new TestLibuvTransportContext() { ConnectionHandler = mockConnectionHandler };
+            var transport = new LibuvTransport(mockLibuv, transportContext, null);
+            var thread = new LibuvThread(transport);
 
-                try
+            try
+            {
+                await thread.StartAsync();
+                await thread.PostAsync(_ =>
                 {
-                    await thread.StartAsync();
-                    await thread.PostAsync(_ =>
+                    var listenerContext = new ListenerContext(transportContext)
                     {
-                        var listenerContext = new ListenerContext(transportContext)
-                        {
-                            Thread = thread
-                        };
-                        var socket = new MockSocket(mockLibuv, Thread.CurrentThread.ManagedThreadId, transportContext.Log);
-                        var connection = new LibuvConnection(listenerContext, socket);
-                        connection.Start();
-
-                        LibuvFunctions.uv_buf_t ignored;
-                        mockLibuv.AllocCallback(socket.InternalGetHandle(), 2048, out ignored);
-                        mockLibuv.ReadCallback(socket.InternalGetHandle(), 0, ref ignored);
-                    }, (object)null);
-
-                    var readAwaitable = await mockConnectionHandler.Input.Reader.ReadAsync();
-                    Assert.False(readAwaitable.IsCompleted);
-                }
-                finally
+                        Thread = thread
+                    };
+                    var socket = new MockSocket(mockLibuv, Thread.CurrentThread.ManagedThreadId, transportContext.Log);
+                    var connection = new LibuvConnection(listenerContext, socket);
+                    _ = connection.Start();
+
+                    mockLibuv.AllocCallback(socket.InternalGetHandle(), 2048, out var ignored);
+                    mockLibuv.ReadCallback(socket.InternalGetHandle(), 0, ref ignored);
+                }, (object)null);
+
+                var readAwaitable = await mockConnectionHandler.Input.Reader.ReadAsync();
+                Assert.False(readAwaitable.IsCompleted);
+            }
+            finally
+            {
+                await thread.StopAsync(TimeSpan.FromSeconds(1));
+            }
+        }
+
+        [Fact]
+        public async Task ConnectionDoesNotResumeAfterSocketCloseIfBackpressureIsApplied()
```

Comment:
Does this test actually consume the "request" data in such a way that the old code would have even tried to resume? Looking at the test, it seems like it would pass in dev since (await flushTask).IsCompleted would already be true.